### PR TITLE
add preparation data for TREC data set

### PR DESCRIPTION
**Pull request by thiziri**
I've added all modules for processing TREC dataset, mainly: the modifications enable to get [TREC](https://trec.nist.gov/trec_eval/) like run with corresponding ids for queries and documents. Hence, the evaluation with trec_eval is possible now. In addition to performing n-cross validation with MatchZoo.
Soon, I'll add programs for constructing TREC input files that are needed by the added functions.

### CODE REVIEW

File: examples/trec/norm_embed.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+if __name__ == '__main__':
+    infile = sys.argv[1]
+    outfile = sys.argv[2]
+    fout = codecs.open(outfile,'w', encoding='utf8')
+    with codecs.open(infile, 'r', encoding='utf8') as f:
+        for line in f:
+            r = line.split()
+            w = r[0]
+            try:
+                # BUG: it will happen `name@domain.com`
```
Comment (bwanglzu) [inline examples/trec/norm_embed.py]: What's this?

File: examples/curve_loss_of_training.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+'''
+import os
+import sys
+import matplotlib.pyplot as plt
+from matplotlib import rcParams
+
+def draw_train_learning_curve(log_file, log_label):
+    #command = 'curl ' + log_file 
+    #r = os.popen(command)
+    r = open(log_file, 'r')
```
Comment (bwanglzu) [inline examples/curve_loss_of_training.py]: Did you close `r` at somewhere? I didn't see it. Why not `with open(file, 'r') as ... `

**Comment by bwanglzu** [comment]
Thanks for your contribution @thiziri . In general, I feel the code is difficult to read. The coding style seems from different people...I don't know. But never mind if this is using for script case.

I would suggest you try to read & practice PEP8 in the future.

**Comment by thiziri** [comment]
okay, I'll do it and push the file again.
Thanks for your review.

**Comment by thiziri** [comment]
I've just made the required changes in my code. Would you like to merge it, please?
Thanks

**Comment by bwanglzu** [comment]
@thiziri Let's wait for the other reviewers :)

**Comment by thiziri** [comment]
Ok, Thanks :+1:

**Comment by thiziri** [comment]
Is it possible to merge it with the master to get my repo up to date?
It seems that there have been several bug fixes in the master.
Thanks

**Comment by bwanglzu** [comment]
@thiziri I'll contact @faneshion on phone :) he's having a vacation.

**Comment by faneshion** [comment]
Sorry for the late responds. Thanks for your contribution. I noticed that there are some modification in files from 'data/toy_example/ranking/'. Now these files are automatically generated by running 'examples/toy_example/generate_ranking_data.sh', and they are fixed. Maybe you can checkout these files.

**Comment by thiziri** [comment]
Hi, 
@faneshion and @yangliuy ,  I've just reverted to the last version in "data/toy_example/ranking", is there any other changes that I had to make? 
I'm waiting to get my repo merged with the [master](https://github.com/faneshion/MatchZoo) in order to be updated with latest fixed errors.
Thanks

**Comment by faneshion** [comment]
I have fully checked your submission, and found your commit is far behind our current master branch. The data preparation code should be in the MatchZoo/data/trec/ directory. All model configuration files as well as model weight files put into the MatchZoo/examples/trec/. By the way, the dataset scripts are also outdated, you can refer to [run_data.sh](https://github.com/faneshion/MatchZoo/blob/master/data/WikiQA/run_data.sh) of WikiQA.

**Comment by thiziri** [comment]
Hi @faneshion @yangliuy 
I've just committed last changes, I created the required scripts for preparing and running on TREC data, I made the asked changes. 
Would you like to check it, please?
I'm waiting for your response
Thanks

**Comment by thiziri** [comment]
Hi, @faneshion @bwanglzu @yangliuy,
Still waiting for a response, are there things that I have to change? or my repo is ready for merging? I would like to be updated with the latest version and with all fixed bugs, I'm using MatchZoo a lot, so I have to be sure of having the latest version, In order to conduct experiments correctly.  
Thanks for understanding

**Comment by bwanglzu** [comment]
Yesterday we discussed about such scripts.

**Comment by bwanglzu** [comment]
> We haven't discussed how to download such data yet, but I would not create a data directory and use some scripts there.

Checkout gluon-nlp, they provided some elegant ways to create data set. Also, keras provided a dataset package to do such things.

I hope 2.0 is able to get rid of all the scripts, and make everything clean & programmable. Also, adding scripts will further reduce our code quality (we miss unit testing, PEP8, documentation) and increase technique debts.

**Comment by thiziri** [comment]
Hi
 @bwanglzu , @faneshion, @yangliuy, and @pl8787 
I would like to continue my experiments with the latest version of MatchZoo, I'm still waiting for a response, is it possible to get my repo merged with the master? otherwise, what should I do?  
Thanks for understanding

**Comment by bwanglzu** [comment]
@thiziri We're going to give up the current `master` branch in the near future, and concentrate on 2.0 branch development.
This means that all the code in master branch will be abandoned in at most 2 months.

**Pull request closed without merging**